PeerioTechnologies/peerio-icebear
# fix(download): wait for file to close before returning and ensure we catch close error

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  

https://peerio.tpondemand.com/entity/7633

#### Linked PRs in other repos
None

#### Breaking changes  

None

#### Manual testing instructions  

Download a file and make sure its preview (or later when desktop is changed, the folder it's in) opens  after it's completely downloaded. (Previously it may have opened before it's downloaded.)

#### Additional notes
... 

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Are there any bicycles invented that should be replaced with existing code or external libs?
* Are there unit and e2e tests added?

----
### Repository owner

- [ ] Breaking changes are applied correctly.
